src/spikeinterface/extractors/nwbextractors.py

```diff
@@ -176,92 +162,100 @@ def __init__(
             sampling_frequency = 1.0 / np.median(np.diff(timestamps[:samples_for_rate_estimation]))
 
         if load_time_vector and timestamps is not None:
-            times_kwargs = dict(time_vector=self._es.timestamps)
+            times_kwargs = dict(time_vector=electrical_series.timestamps)
         else:
             times_kwargs = dict(sampling_frequency=sampling_frequency, t_start=t_start)
 
-        num_frames = int(self._es.data.shape[0])
-
         # Extractors channel groups must be integers, but Nwb electrodes group_name can be strings
-        if "group_name" in self._nwbfile.electrodes.colnames:
-            unique_grp_names = list(np.unique(self._nwbfile.electrodes["group_name"][:]))
+        if "group_name" in electrodes_table.colnames:
+            unique_grp_names = list(np.unique(electrodes_table["group_name"][:]))
 
         # Fill channel properties dictionary from electrodes table
-        if "channel_name" in self._nwbfile.electrodes.colnames:
-            channel_ids = [self._es.electrodes["channel_name"][i] for i in self._es.electrodes.data]
+        if "channel_name" in electrodes_table.colnames:
+            channel_ids = [electrical_series.electrodes["channel_name"][i] for i in electrodes_indices]
```

Comment:
I've also been getting more in the habit of using better list compression names, rather than mathematical variables

```suggestion
            channel_ids = [electrical_series.electrodes["channel_name"][electrodes_index] for electrodes_index in electrodes_indices]
```

Reply:
Yes, this is better.